support ET dump for llama3 runner

billmguo opened the pull request:
Summary: Support ET dump for llama3 runner to easy understand the regression performance issue

Differential Revision: D67656207

cccclai commented on examples/qualcomm/oss_scripts/llama3_2/runner/runner.h:
```diff
@@ -93,6 +97,11 @@ class Runner {
+  torch::executor::ETDumpGen*  prefill_dump_ = nullptr;
+  torch::executor::ETDumpGen*  decode_dump_ = nullptr;
+  bool gen_etdump_ = false;
+  std::string prefill_etdump_path_;
```
Can they can be dumped to the .pte file folder

billmguo replied:
update with models_path[0] dir

haowhsu-quic commented on examples/qualcomm/oss_scripts/llama3_2/runner/runner.cpp:
```diff
@@ -54,6 +55,27 @@ Runner::Runner(
+    gen_etdump_ = true;
+    switch(eval_mode) {
+      case EvalMode::kPrefill:
+        prefill_dump_ = new torch::executor::ETDumpGen();
```
Maybe use smart pointer to take care of life cycle.

billmguo replied:
update the diff with smart pointer

billmguo committed:
support ET dump for llama3 runner (#7507)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/7507

Support ET dump for llama3 runner to easy understand the regression performance issue

Reviewed By: Andriyluck, limintang

Differential Revision: D67656207
